### PR TITLE
Improvement/startup performance

### PR DESCRIPTION
**Pull request by maximilianmeier**
### Description
Improved the startup of the app by removing 3 rest api calls. the functionality is not affected by this

### CODE REVIEW

File: src/main/webapp/app/layouts/main/main.component.ts [hunk 1]
```diff
@@ -22,15 +23,17 @@ export class JhiMainComponent implements OnInit {
...
-        if (profileInfo.inProduction && window.location.host === 'artemis.ase.in.tum.de') {
-            // only Track in Production Environment
-            this.angularticsPiwik.startTracking();
-        } else {
-            // Enable Developer Mode in all other environments
-            this.angulartics.settings.developerMode = true;
-        }
+        this.profileService.getProfileInfo().subscribe((profileInfo: ProfileInfo) => {
+            if (profileInfo) {
+                if (profileInfo.inProduction && window.location.host === 'artemis.ase.in.tum.de') {
```
Comment (jpbernius) [inline src/main/webapp/app/layouts/main/main.component.ts]: I think we can simplify this and remove one level of `if` conditions. Something like this maybe?
```suggestion
                if (profileInfo && profileInfo.inProduction && window.location.host === 'artemis.ase.in.tum.de') {
```

File: src/main/webapp/app/components/course-registration-selector/course-registration-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import { Course, CourseService } from 'app/entities/course';
+import { HttpResponse } from '@angular/common/http';
+
+const DEFAULT_COLORS = ['#6ae8ac', '#9dca53', '#94a11c', '#691b0b', '#ad5658', '#1b97ca', '#0d3cc2', '#0ab84f'];
+
+@Component({
+    selector: 'jhi-course-registration-selector',
+    templateUrl: './course-registration-selector.component.html',
+})
+export class CourseRegistrationSelectorComponent implements OnInit {
```
Comment (jpbernius) [inline src/main/webapp/app/components/course-registration-selector/course-registration-selector.component.ts]: Good Refactoring! 👍

File: src/main/webapp/app/sentry/sentry.error-handler.ts [hunk 3]
```diff
@@ -19,16 +19,18 @@ export class SentryErrorHandler implements ErrorHandler {
     }
 
     private async initSentry(): Promise<void> {
-        const profileInfo = await this.profileService.getProfileInfo();
+        this.profileService.getProfileInfo().subscribe((profileInfo: ProfileInfo) => {
+            if (profileInfo) {
+                if (!profileInfo.sentry) {
```
Comment (jpbernius) [inline src/main/webapp/app/sentry/sentry.error-handler.ts]: Can we simplify this change with something like this:
```suggestion
                if (! profileInfo || !profileInfo.sentry) {
```

File: src/main/webapp/app/layouts/main/main.component.ts (hunk 1 above)
Comment (maximilianmeier) [reply src/main/webapp/app/layouts/main/main.component.ts]: good point. i'll change it

File: src/main/webapp/app/sentry/sentry.error-handler.ts (hunk 3 above)
Comment (maximilianmeier) [reply src/main/webapp/app/sentry/sentry.error-handler.ts]: good point. i'll change it

**Pull request merged**